### PR TITLE
C#: Introduce log verbosity extractor option 

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR adds a new extractor option to control the log verbosity in the C# extractor. If not specified, we fall back to the `CODEQL_VERBOSITY` environment variable, which is not yet set in the CLI, but it will. If that's not set, then we use the previously used `--verbosity`, `--verbose` or `--silent` options. Finally if none of these are set, then we set the verbosity to `info`.

The autobuilder is using `CODEQL_VERBOSITY` or if not set, it falls back to `info`.

Additionally, the PR does some refactoring around logging. Commit-by-commit review is suggested.

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction/Options.cs [hunk 1]
```diff
@@ -20,6 +20,36 @@ public abstract class CommonOptions : ICommandLineOptions
         /// </summary>
         public Verbosity LegacyVerbosity { get; protected set; } = Verbosity.Info;
```
Comment (michaelnebel) [inline csharp/extractor/Semmle.Extraction/Options.cs:21]: Can this be made private?

**Review by michaelnebel — APPROVED** [review]
This looks good to me!
Does it make sense to mention the precedence of the logging flags in the change note?
(1) C# extractor option
(2) CodeQL CLI option
(3) Legacy flag.

File: csharp/extractor/Semmle.Extraction/Options.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction/Options.cs:21]: Unit tests are still accessing this property, so it can't be made `private`.

**Comment by tamasvajk** [comment]
> This looks good to me! Does it make sense to mention the precedence of the logging flags in the change note? (1) C# extractor option (2) CodeQL CLI option (3) Legacy flag.

I don't think it's worth mentioning. The CLI verbosity is not yet set, it doesn't exists currently. (The C# implementation follows the Python one, which also adds support for it.) Also, I wouldn't mention the legacy flag, because that's not available publicly. It's only set by the test runner when `codeql test run` directly invokes the extractor.

**Pull request merged**